src/engine/tools/engine-suite/src/engine_test/api_connector.py

```diff
@@ -0,0 +1,104 @@
+from datetime import datetime
+from enum import Enum
+from api_communication import communication
+from api_communication import test_pb2
+
+class ApiConfig(Enum):
+    OriginName = "engine-test"
+    OriginModule = "engine-test"
+    Socket = "/var/ossec/queue/sockets/engine-api"
+    Component = "test"
+    SessionName = "engine_test"
+    Lifespan = 0
+    Description = ""
+
+class ApiConnector:
+    def __init__(self, config):
+        try:
+            self.api_client = communication.APIClient(ApiConfig.Socket.value, ApiConfig.Component.value)
+            self.session_name = ApiConfig.SessionName.value
+        except Exception as ex:
+            print('Could not establish communication with the API. Error: {}'.format(ex))
+            exit(1)
+
+        self.config = config
+        self.create_session()
+
+    def send_event(self, event):
+        try:
+            # Create test request
+            request = test_pb2.RunPost_Request()
+            request.name = self.session_name
+            request.protocol_queue = chr(self.config['queue'])
+            request.protocol_location = self.config['origin']
+            request.event.string_value = event
+
+            request.namespaces.extend(self.config['namespaces'])
+
+            if (self.config['full_verbose'] == True):
+                request.debug_mode = 2
+            else:
+                if (self.config['verbose'] == True):
+                    request.debug_mode = 1
+
+            if self.config['assets']:
+                request.asset_trace.extend(self.config['assets'])
+
+            print("\nSent: \n{}".format(request))
+            response = self.api_client.send_command("run", "post", request)
+            print("Received: \n{}".format(response))
+            return response
+        except Exception as ex:
+            print('Could not send event to TEST api. Error: {}'.format(ex))
+            exit(2)
+
+    def create_session(self):
+        try:
+            if self.config['session_name']:
```

Comment:
questing - desing: because you need the name of the session if you already have it hardcoded in class ApiConfig(Enum)

Reply:
The session can be parameterized by the CLI, so it is validated if you want to use a particular session or if the default one will be used.